packages/react-core/src/components/InlineEdit/InlineEdit.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import styles from '@patternfly/react-styles/css/components/InlineEdit/inline-edit';
+import { css } from '@patternfly/react-styles';
+import { InlineEditActionGroupProps } from './InlineEditActionGroup';
+import { InlineEditToggle } from './InlineEditToggle';
+
+export interface InlineEditProps {
+  /** Determines whether inline edit is in edit mode. */
+  isEditMode?: boolean;
+  /** Element rendered when inline edit is not in edit mode. */
+  staticElement?: React.ReactNode;
+  /** Element rendered when inline edit is in edit mode. */
+  editModeElement?: React.ReactNode;
+  /** Action to perform on the edit icon click. It should set isEditMode prop to true. */
+  onEditToggle?: (event: React.MouseEvent) => void;
+  /** Element consisting of "save" and "cancel" buttons. InlineEditActionGroup component should be used. */
+  actionGroup?: React.ReactElement<InlineEditActionGroupProps>;
+}
+
+export const InlineEdit: React.FunctionComponent<InlineEditProps> = ({
+  isEditMode,
+  staticElement,
+  editModeElement,
+  onEditToggle,
+  actionGroup
+}: InlineEditProps) => (
+  <form className={css(styles.inlineEdit, isEditMode && styles.modifiers.inlineEditable)}>
```

Comment:
@thatblindgeye good question - it looks like the core examples aren't consistent, either. My guess is the inputs/buttons need a wrapper, and a `<form>` is a semantic wrapper for the inputs/buttons, so it's an option, and may be better than something generic like a `<div>` for those examples - especially for a core user that may not be using react and might handle form submissions in a more traditional/server-side way. And AFAIK it will also wire up hitting enter to submit the form, though the event handler will need to stop the form from submitting and refreshing the page.

I guess it's a valid _option_ for wrapping the editable thing/buttons, though as a default or in what context in a react app, I'd probably defer to y'all. I'd be down to chat about it if you'd like.

Reply:
For now I added a prop to customize the wrapper element, defaulting to "form". I had been thinking that within the context of a table row we may want a `tr` wrapper for semantics, maybe some other element in some other context, etc.